Misc/NEWS.d/next/Windows/2022-12-06-11-16-46.gh-issue-99941.GmUQ6o.rst

```diff
@@ -0,0 +1,6 @@
+Made :func:`asyncio.Protocol.data_received` receive immutable object
+``bytes`` instead of ``bytearrays`` with regression tests added and opted to
+use ``memoryview`` to take the slices for performance on big buffers.
+Performance test from ``gh-issue:21442`` done on my machine with 10
+repetitions on release build with timings 0.043 (delta: 0.019) before patch
+and 0.041 (delta: 0.010) after so no noticable change in performance.
```

Comment:
```suggestion
Ensure that :func:`asyncio.Protocol.data_received` receives an immutable
:class:`bytes` object (as documented), instead of :class:`bytearray`.
```

The implementation strategy isn't important for the news entry, nor is the perf info.
Those could go into the PR description and/or in the commit message though.